dhall/CHANGELOG.md

```diff
@@ -1,3 +1,44 @@
+1.30.0
+
+* [Supports version 14.0.0 of the standard](https://github.com/dhall-lang/dhall-lang/releases/tag/v14.0.0)
+* BREAKING CHANGE TO THE API: [Add `--check` flag to `dhall {lint,freeze}`](https://github.com/dhall-lang/dhall-haskell/pull/1636)
+    * You can now use the `--check` flag to verify that a file has already been
+      linted or frozen
+    * This is a breaking change to the types used by the `Dhall.Format` module
+* BREAKING CHANGE TO THE LANGUAGE: [Disallow `Natural` literals with leading zeros](https://github.com/dhall-lang/dhall-haskell/pull/1658)
+    * Now a literal like `042` is no longer valid
+    * See the [changelog for standard version 14.0.0](https://github.com/dhall-lang/dhall-lang/releases/tag/v14.0.0) for more details
+* BUG FIX: [Fix parsing of `Double` literal trailing whitespace](https://github.com/dhall-lang/dhall-haskell/pull/1647)
+    * Certain expressions using `Double` literals would fail to parse, which this
+      change fixes
+* BUG FIX: [Use `DeriveLift` instead of GHC Generics to derive `Lift` ](Use `DeriveLift` instead of GHC Generics to derive `Lift` )
+    * This fixes a build failure on GHC 8.6 and GHC 8.8 when building with `-O2`
```

Comment:
I believe Ryan had encountered the issue with the empty `Lift` instances with GHC 8.10, but not with 8.6 or 8.8.

FWIW I had never reproduced _that_ issue myself.

The fix for the panic with `-O2` (https://github.com/dhall-lang/dhall-haskell/issues/1639) is still WIP on the GHC side.

(Also, we ended up defining `lift` via `liftData` instead of using DeriveLift.)

Reply:
…and it seems like we should probably go with `DeriveLift` eventually! If I compile `dhall` with 8.10-rc1 I get warnings like this one:

```
src/Dhall/Syntax.hs:123:10: warning: [-Wmissing-methods]
    • No explicit implementation for
        ‘Syntax.liftTyped’
    • In the instance declaration for ‘Lift Const’
    |
123 | instance Lift Const where
    |          ^^^^^^^^^^
```

[`liftTyped` is the new minimal complete definition in `template-haskell-2.16`.](http://downloads.haskell.org/~ghc/8.10.1-rc1/docs/html/libraries/template-haskell-2.16.0.0/Language-Haskell-TH-Syntax.html#t:Lift)